internal/loader/settings_client.go

```diff
@@ -11,26 +11,78 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
-	"k8s.io/klog/v2"
+	"k8s.io/klog"
 )
 
 //go:generate mockgen -destination=mocks/mock_settings_client.go -package mocks . SettingsClient
 
-type SelectorSettingsClient struct {
-	selectors []acpv1.Selector
+type EtagSettingsClient struct {
+	etags       map[acpv1.Selector][]*azcore.ETag
+	etagChanged bool
 }
 
 type SentinelSettingsClient struct {
 	sentinel        acpv1.Sentinel
 	etag            *azcore.ETag
 	refreshInterval string
 }
+type SelectorSettingsClient struct {
+	selectors []acpv1.Selector
+}
 
 type SettingsClient interface {
-	GetSettings(ctx context.Context, client *azappconfig.Client) ([]azappconfig.Setting, error)
+	GetSettings(ctx context.Context, client *azappconfig.Client) ([]azappconfig.Setting, map[acpv1.Selector][]*azcore.ETag, error)
```

Comment:
I don't really the return type in an interface method, could we create a new type for them?